optimize/backend/src/main/java/io/camunda/optimize/TomcatConfig.java

```diff
@@ -0,0 +1,273 @@
+    final ResponseHeadersFilter responseHeadersFilter =
+        new ResponseHeadersFilter(configurationService);
+    final FilterRegistrationBean<ResponseHeadersFilter> registrationBean =
+        new FilterRegistrationBean<>();
+    registrationBean.addUrlPatterns("/*");
+    registrationBean.setFilter(responseHeadersFilter);
+    return registrationBean;
+  }
+
+  public int getPort(final String portType) {
+    final String portProperty = environment.getProperty(portType);
+    if (portProperty != null) {
+      try {
+        return Integer.parseInt(portProperty);
+      } catch (final NumberFormatException exception) {
+        throw new OptimizeConfigurationException("Error while determining container port");
+      }
+    }
+
+    if (portType.equals(EnvironmentPropertiesConstants.HTTPS_PORT_KEY)) {
+      return configurationService.getContainerHttpsPort();
+    }
+
+    final Optional<Integer> httpPort = configurationService.getContainerHttpPort();
+    if (httpPort.isEmpty()) {
+      throw new OptimizeConfigurationException("HTTP port not configured");
+    }
+    return httpPort.get();
+  }
+
+  public Optional<String> getContextPath() {
+    // If the property is set by env var (the case when starting a new Optimize in ITs), this takes
+    // precedence over config
+    final Optional<String> contextPath = Optional.ofNullable(environment.getProperty(CONTEXT_PATH));
+    if (contextPath.isEmpty()) {
+      return configurationService.getContextPath();
+    }
+    return contextPath;
+  }
+
+  private SSLHostConfig getSslHostConfig() {
+    final SSLHostConfig sslHostConfig = new SSLHostConfig();
+    sslHostConfig.setHostName(configurationService.getContainerHost());
+
+    final SSLHostConfigCertificate cert =
+        new SSLHostConfigCertificate(sslHostConfig, Type.UNDEFINED);
+    cert.setCertificateKeystoreFile(configurationService.getContainerKeystoreLocation());
+    cert.setCertificateKeystorePassword(configurationService.getContainerKeystorePassword());
+    sslHostConfig.addCertificate(cert);
+
+    return sslHostConfig;
+  }
+
+  private void enableGzipSupport(final Connector connector) {
+    connector.setProperty("compression", "on");
+    connector.setProperty("compressionMinSize", "23");
+    connector.setProperty("compressionNoCompressionMethods", ""); // all methods
+    connector.setProperty("useSendfile", "false");
+    connector.setProperty("compressableMimeType", String.join(",", COMPRESSED_MIME_TYPES));
+  }
+
+  private void setMaxHeaderSize(final Connector connector) {
+    // NOTE: In Tomcat, the request and response header size are both controlled
+    // by a single property called maxHeaderSize.
+    final int maxHeaderSize =
+        Math.max(
+            configurationService.getMaxResponseHeaderSizeInBytes(),
+            configurationService.getMaxRequestHeaderSizeInBytes());
+    connector.setProperty("maxHeaderSize", String.valueOf(maxHeaderSize));
```

Comment:
probably a good one for @matthewBearCamunda to test.

I am not sure I like the approach of taking the max value for both here. I think a better solution would be to either just pick one and we delete the other property (documenting that we are doing so). Alternatively, we introduce a new property to better capture this behaviour.

In any case, you can create a separate issue to revisit this in a follow-up if you like

Reply:
Yeah, I forgot to comment here.

For the moment, I opted not to change the configuration and just use it the best way I could, but my preference would be to change the documentation and use a variable that corresponds to what Tomcat uses, that is, one single variable that controls the min and the max size.

I decided not to do it now just because it implies a doc change. I am going to file a task to do it later, thanks.

Reply:
Done:
https://github.com/camunda/camunda/issues/24629

Reply:
I think this goes away thanks to Matt's comment below

Reply:
@buccarel - I'd still want us to manually test out the header size change